From: Daniel Burrows <Daniel_Burrows@alumni.brown.edu>
Date: Tue, 5 Apr 2011 21:58:01 -0700
Subject: Comment out a backwards compatibility typedef

This typedef doesn't work with g++ >= 4.3, and it's not required on Linux.
---
 sigc++/signal.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/sigc++/signal.h b/sigc++/signal.h
index 386b590..251d1ff 100644
--- a/sigc++/signal.h
+++ b/sigc++/signal.h
@@ -25,7 +25,11 @@
   //Compilers, such as older versions of SUN Forte C++, that do not allow this also often
   //do not allow a typedef to have the same name as a class in the typedef's definition.
   //For Sun Forte CC 5.7 (SUN Workshop 10), comment this out to fix the build.
-  #define SIGC_TYPEDEF_REDEFINE_ALLOWED 1
+  //
+  // Debian-specific patch: gcc-4.3 does not allow this either, and the
+  // comments below say this is a backwards-compatibility typedef, so
+  // just don't enable this feature at all.
+  //#define SIGC_TYPEDEF_REDEFINE_ALLOWED 1
 #endif
 
 namespace sigc {
